packages/lib-utils/src/app/redux/actions/k8s/index.ts

```diff
@@ -0,0 +1,335 @@
+import type { DiscoveryResources } from '../../../../types/api-discovery';
+import type { K8sModelCommon, K8sResourceCommon, FilterValue } from '../../../../types/k8s';
+import { getImpersonate, getActiveCluster } from '../../reducers/core';
+
+export enum ActionType {
+  ReceivedResources = 'resources',
+  GetResourcesInFlight = 'getResourcesInFlight',
+  StartWatchK8sObject = 'startWatchK8sObject',
+  StartWatchK8sList = 'startWatchK8sList',
+  ModifyObject = 'modifyObject',
+  StopWatchK8s = 'stopWatchK8s',
+
+  Errored = 'errored',
+  Loaded = 'loaded',
+  BulkAddToList = 'bulkAddToList',
+  UpdateListFromWS = 'updateListFromWS',
+  FilterList = 'filterList',
+}
+
+type K8sEvent = { type: 'ADDED' | 'DELETED' | 'MODIFIED'; object: K8sResourceCommon };
+
+export const updateListFromWS = (id: string, k8sObjects: K8sEvent[]) =>
+  action(ActionType.UpdateListFromWS, { id, k8sObjects });
+export const loaded = (id: string, k8sObjects: K8sResourceCommon | K8sResourceCommon[]) =>
+  action(ActionType.Loaded, { id, k8sObjects });
+
+export const bulkAddToList = (id: string, k8sObjects: K8sResourceCommon[]) =>
+  action(ActionType.BulkAddToList, { id, k8sObjects });
+
+export const startWatchK8sObject = (id: string) => action(ActionType.StartWatchK8sObject, { id });
+export const startWatchK8sList = (id: string, query: { [key: string]: string }) =>
+  action(ActionType.StartWatchK8sList, { id, query });
+export const modifyObject = (id: string, k8sObjects: K8sResourceCommon) =>
+  action(ActionType.ModifyObject, { id, k8sObjects });
+export const stopWatchK8s = (id: string) => action(ActionType.StopWatchK8s, { id });
+
+export const errored = (id: string, k8sObjects: unknown) =>
+  action(ActionType.Errored, { id, k8sObjects });
+export const filterList = (id: string, name: string, value: FilterValue) =>
+  action(ActionType.FilterList, { id, name, value });
+
+export const partialObjectMetadataListHeader = {
+  Accept: 'application/json;as=PartialObjectMetadataList;v=v1;g=meta.k8s.io,application/json',
+};
+
+export const partialObjectMetadataHeader = {
+  Accept: 'application/json;as=PartialObjectMetadata;v=v1;g=meta.k8s.io,application/json',
+};
+
+// TODO remove prolific use of any type
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const WS = {} as { [id: string]: WebSocket & any };
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const POLLs: any = {};
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const REF_COUNTS: any = {};
+
+const paginationLimit = 250;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const stopK8sWatch = (id: string) => (dispatch: any) => {
+  REF_COUNTS[id] -= 1;
+  if (REF_COUNTS[id] > 0) {
+    return _.noop;
+  }
+
+  const ws = WS[id];
+  if (ws) {
+    ws.destroy();
+    delete WS[id];
+  }
+  const poller = POLLs[id];
+  clearInterval(poller);
+  delete POLLs[id];
+  delete REF_COUNTS[id];
+  return dispatch(stopWatchK8s(id));
```

Comment:
We have a linter rule that prevents this: `consistent-return`
<img width="749" alt="image" src="https://user-images.githubusercontent.com/21317056/157073448-58f15f29-8903-4d70-839c-e3a7137bc6e7.png">

Reply:
Also some conflicting calls below to `stopK8sWatch` - we need to look at better typing for thunks

Reply:
Thunk problem is here: https://github.com/openshift/dynamic-plugin-sdk/pull/37#discussion_r819868447

Reply:
`Consistent-return` is because you have another return statement earlier that is returning a value -- at least I'm fairly sure... I think it's just saying "return always"